### PR TITLE
FDTN-471: Investigate location response time spikes

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: api/app/controllers/Addresses.scala [hunk 1]
```diff
@@ -47,10 +50,15 @@ class Addresses @javax.inject.Inject() (
           case Left(error) => {
             logger
               .withKeyValue("address", text)
-              .withKeyValue("error_code", error.code)
+              .withKeyValue("error_code", error.code.toString)
```
Comment (benwaffle) [inline api/app/controllers/Addresses.scala:53]: toString unecessary here

File: deploy/location/templates/deployment.yaml [hunk 2]
```diff
@@ -39,7 +39,7 @@ spec:
           imagePullPolicy: {{ $.Values.image.pullPolicy }}
           env:
             - name: JAVA_OPTS
-              value: "-Xms{{ $.Values.jvmMemory}}m -Xmx{{ $.Values.jvmMemory}}m -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/opt -XX:FlightRecorderOptions=stackdepth=256"
+              value: "-Xms{{ $.Values.jvmMemory}}m -Xmx{{ $.Values.jvmMemory}}m -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/opt -XX:FlightRecorderOptions=stackdepth=256 -XX:MaxGCPauseMillis=100 -XX:+UseStringDeduplication -XX:G1ReservePercent=20 -XX:InitiatingHeapOccupancyPercent=35"
```
Comment (benwaffle) [inline deploy/location/templates/deployment.yaml:42]: Can you just double check on your laptop that these flags work on java 13?

**Review by benwaffle — CHANGES_REQUESTED** [review]

File: api/app/controllers/Addresses.scala (hunk 1 above)
Comment (benwaffle) [reply api/app/controllers/Addresses.scala:53]: in fact, we could even just do `.withKeyValue("error", error)` here

Comment (gheine) [reply api/app/controllers/Addresses.scala:53]: No it's not. Otherwise it'll json serialize the code object.

Comment (benwaffle) [reply api/app/controllers/Addresses.scala:53]: Yup, that's the idea

File: deploy/location/templates/deployment.yaml (hunk 2 above)
Comment (gheine) [reply deploy/location/templates/deployment.yaml:42]: 👍  they do :-)

File: api/app/controllers/Addresses.scala (hunk 1 above)
Comment (gheine) [reply api/app/controllers/Addresses.scala:53]: Meh, seems unnecessary in this case

File: deploy/location/templates/deployment.yaml (hunk 2 above)
Comment (gheine) [reply deploy/location/templates/deployment.yaml:42]: If this change isn't having the desired effect, we could try ZGC instead of G1

Comment (benwaffle) [reply deploy/location/templates/deployment.yaml:42]: see my slack DMs

**Pull request merged**